extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -0,0 +1,661 @@
+                            });
+                            return next(err);
+                        }
+                        const objDate = new Date(data.LastModified);
+                        const now = Date.now();
+                        // time diff in ms
+                        const diff = now - objDate;
+                        const daysSinceCR =
+                            Math.floor(diff / (1000 * 60 * 60 * 24));
+                        const ncve = 'NoncurrentVersionExpiration';
+                        if (applicableRules[ncve].NoncurrentDays &&
+                        applicableRules[ncve].NoncurrentDays <= daysSinceCR) {
+                            this.sendObjectEntry({
+                                action: 'delete',
+                                target: {
+                                    owner: bucketData.owner,
+                                    bucket: bucketData.bucket,
+                                    key: version.Key,
+                                    version: data.VersionId,
+                                },
+                            });
+                        }
+                        return next();
+                    });
+                },
+            ], err => cb(err));
+        }, done);
+    }
+
+    // _compareDeleteMarkers(bucket, rules, deleteMarkers, versions, log, cb) {
+    //      // will be completed later
+    // }
+
+    _compareWithLCRules(bucketData, bucketLCRules, data, log, done) {
+        async.parallel([
+            next => this._compareContents(bucketData, bucketLCRules,
+                data.Contents, log, next),
+            next => this._compareVersions(bucketData, bucketLCRules,
+                data.Versions, log, next),
+            next => this._compareDeleteMarkers(bucketData, bucketLCRules,
+                data.DeleteMarkers, data.Versions, log, next),
+        ], done);
+    }
+
+    /**
+     * Process a single bucket entry with lifecycle configurations enabled
+     * @param {array} bucketLCRules - array of bucket lifecycle rules
+     * @param {object} bucketData - bucket data from zookeeper bucketTasksTopic
+     * @param {string} bucketData.bucket - name of bucket
+     * @param {string} bucketData.ownerID - owner id
+     * @param {string} [bucketData.prefix] - prefix
+     * @param {string} [bucketData.nextKeyMarker] - next key marker for
+     *   versioned buckets
+     * @param {string} [bucketData.nextVersionIdMarker] - next version id marker
+     *   for versioned buckets
+     * @param {string} [bucketData.nextContinuationToken] - next continuation
+     *   token marker for non-versioned buckets
+     * @param {function} done - callback(error, data)
+     * @return {undefined}
+     */
+    processBucketEntry(bucketLCRules, bucketData, done) {
+        const log = this.log.newRequestLogger();
+
+        log.debug('processing bucket entry', {
+            bucket: bucketData.bucket,
+            owner: bucketData.ownerId,
```

Comment:
Based on the design docs, and what I have in the producer, `bucketData` should have an `owner` property rather than `ownerId`. Either way works for me, but we should just settle on one.